src/components/TodoList.tsx

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { useContext } from 'react';
+import { MyContext } from './state';
+import { Filter } from '../Types/Filter';
+import React from 'react';
+import { Todo } from '../Types/Todo';
+
+export const TodoList: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    filter,
+    deleteTodo,
+    setEditingTodoId,
+    editingTodoId,
+  } = useContext(MyContext);
+
+  const filteredTodos = todos.filter((todo: Todo) => {
+    switch (filter) {
+      case Filter.ACTIVE:
+        return !todo.completed;
+      case Filter.COMPLETED:
+        return todo.completed;
+      case Filter.ALL:
+        return true;
+      default:
+        return false;
+    }
+  });
+
+  const toggleTodoCompleted = (id: number) => {
+    setTodos((currentTodos: Todo[]) =>
+      currentTodos.map((todo: Todo) =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  };
+
+  const handleUpdateTodo = (id: number, newTitle: string) => {
+    const trimmedTitle = newTitle.trim();
+
+    if (trimmedTitle) {
+      setTodos((currentTodos: Todo[]) =>
+        currentTodos.map((todo: Todo) =>
+          todo.id === id ? { ...todo, title: trimmedTitle } : todo,
+        ),
+      );
+    } else {
+      deleteTodo(id);
+    }
+
+    setEditingTodoId(null);
+  };
+
+  const handleBlur = (id: number, currentTitle: string) => {
+    handleUpdateTodo(id, currentTitle);
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {filteredTodos.map((todo: Todo) => (
+        <div
+          data-cy="Todo"
+          className={`todo ${todo.completed ? 'completed' : ''}`}
```

Comment:
```suggestion
          className={cn('todo', {'completed':todo.completed})}
```
it's looks much better


also if we will need to add some another class here

without classnames:
```
className={`todo ${todo.completed ? 'completed' : ''} ${smth ? 'test-class' : ''}`}
```
with classnames:
```
className={cn('todo', {'completed':todo.completed, 'test-class': smth})}
```